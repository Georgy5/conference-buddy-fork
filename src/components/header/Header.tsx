import React from "react"
import { StaticImage } from "gatsby-plugin-image"

function Header() {
  return (
    <header className="mb-3">
      <nav className="bg-confbuddy-green px-sm-4 py-2 px-2 d-flex justify-content-between">
        <div className="d-none d-sm-block d-flex align-items-center">
          <a href="/" title="Startpage">
            <StaticImage
              src="../../images/LogoConferenceBuddy_head_small.png"
              alt="Logo"
            />
          </a>
        </div>
      </nav>
    </header>
  )
}
export default Header
