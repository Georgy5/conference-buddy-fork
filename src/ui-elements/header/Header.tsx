import React, { ReactElement } from "react"
import { StaticImage } from "gatsby-plugin-image"
import { Link } from "gatsby"
import { SignIn } from "../../domain/sign-in/SignIn"

function Header(): ReactElement {
  return (
    <header className="mb-3">
      <nav className="bg-confbuddy-green px-sm-4 py-2 px-2 d-flex justify-content-between">
        <div className="d-flex align-items-center">
          <Link to="/" title="Startpage">
            <StaticImage
              src="../../assets/images/LogoConferenceBuddy.png"
              alt="Conference Buddy logo"
              height={50}
              loading="eager"
              placeholder="none"
            />
          </Link>
          <div className="text-white d-block d-sm-none">Conference Buddy</div>
          <div className="text-white d-none d-sm-block fs-3">
            Conference Buddy
          </div>
        </div>
        <SignIn />
      </nav>
    </header>
  )
}
export { Header }
