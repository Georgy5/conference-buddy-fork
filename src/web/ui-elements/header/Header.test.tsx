import { render, screen, cleanup, RenderResult } from "@testing-library/react"
import { Header } from "./Header"
import { QueryClient, QueryClientProvider } from "react-query"
import { ReactElement } from "react"

export function renderWithClient(client: QueryClient, ui: ReactElement) {
  const { rerender, ...result } = render(
    <QueryClientProvider client={client}>{ui}</QueryClientProvider>
  )
  return {
    ...result,
    rerender: (rerenderUi: ReactElement) =>
      rerender(
        <QueryClientProvider client={client}>{rerenderUi}</QueryClientProvider>
      ),
  }
}

describe("Header", () => {
  let component: RenderResult
  beforeAll(() => {
    const queryClient = new QueryClient()
    component = renderWithClient(queryClient, <Header />)
  })

  afterAll(cleanup)

  it("renders the header element", () => {
    const header = screen.getByRole("banner")
    expect(header).toBeVisible()
  })

  it("renders all necessary elements", () => {
    expect(component.asFragment()).toMatchSnapshot()
  })
})
