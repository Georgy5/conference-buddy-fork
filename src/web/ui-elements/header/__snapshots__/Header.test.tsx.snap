// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders all necessary elements 1`] = `
<DocumentFragment>
  <header>
    <nav
      class="bg-confbuddy-green px-sm-4 py-2 px-2 d-flex justify-content-between align-items-center"
    >
      <div
        class="d-flex flex-grow-1 align-items-center"
      >
        <a
          href="/"
          title="Startpage"
        >
          <img
            alt="Conference Buddy logo"
            height="50"
            loading="eager"
            placeholder="none"
            src="../../assets/images/LogoConferenceBuddy.png"
          />
        </a>
        <div
          class="text-white d-block d-sm-none d-flex align-items-center"
        >
          Conference Buddy
        </div>
        <div
          class="text-white d-none d-sm-block fs-3"
        >
          Conference Buddy
        </div>
      </div>
      <button
        class="btn btn-outline-light btn-sm"
      >
        Sign In
      </button>
    </nav>
  </header>
</DocumentFragment>
`;
